app/Http/Controllers/Admin/ProductController.php

```diff
@@ -0,0 +1,131 @@
+<?php
+
+namespace App\Http\Controllers\Admin;
+
+use App\Category;
+use App\Product;
+use App\Http\Controllers\Controller;
+use Illuminate\Support\Facades\Auth;
+use App\Http\Requests\ProductRequest;
+use App\Services\ProductService;
+use App\Traits\UploadTrait;
+
+class ProductController extends Controller
+{
+    use UploadTrait;
+
+    protected $productService;
+
+    public function __construct(ProductService $productService)
+    {
+        $this->productService = $productService;
+    }
+
+    /**
+     * Display a listing of the resource.
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function index()
+    {
+        $products = Product::with('categories')->get();
+
+        return view('admin.products.index', compact('products'));
+    }
+
+    /**
+     * Show the form for creating a new resource.
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function create()
+    {
+        $categories = Category::whereNull('parent_id')->get();
+
+        return view('admin.products.create', compact('categories'));
+    }
+
+    /**
+     * Store a newly created resource in storage.
+     *
+     * @param  \Illuminate\Http\Request  $request
+     * @return \Illuminate\Http\Response
+     */
+    public function store(ProductRequest $request)
+    {
+        $data = $request->only([
+            'image.*',
+            'name',
+            'gender',
+            'category',
+            'description',
+            'publish_date',
+            'price',
+            'sale'
+        ]);
+
+        $data['user_id'] = Auth::id();
```

Comment:
```suggestion
        $data['user_id'] = auth()->id();
```
sài helper như này cũng được nè